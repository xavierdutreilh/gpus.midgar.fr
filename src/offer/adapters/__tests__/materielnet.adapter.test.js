jest.mock("node-fetch")

const fetch = require("node-fetch")
const FormData = require("form-data")
const fs = require("fs").promises
const { Response } = jest.requireActual("node-fetch")
const { when } = require("jest-when")

const materielnet = require("../materielnet.adapter")

describe("getOffers", () => {
  beforeAll(async () => {
    when(fetch)
      .calledWith(
        "https://www.materiel.net/recherche/+fcat-426+fv121-19183,19185.html"
      )
      .mockResolvedValue(
        new Response(
          await fs.readFile(
            `${__dirname}/mocks/get-materielnet-offers.html`,
            "utf8"
          )
        )
      )
    when(fetch)
      .calledWith("https://www.materiel.net/product-listing/stock-price/")
      .mockResolvedValue(
        new Response(
          await fs.readFile(
            `${__dirname}/mocks/get-materielnet-prices.json`,
            "utf8"
          )
        )
      )
  })

  it("retrieves materiel.net offers", async () => {
    const offers = await materielnet.getOffers()
    expect(fetch).toHaveBeenCalledTimes(2)
    expect(fetch).toHaveBeenCalledWith(
      "https://www.materiel.net/recherche/+fcat-426+fv121-19183,19185.html"
    )
    expect(fetch).toHaveBeenCalledWith(
      "https://www.materiel.net/product-listing/stock-price/",
      {
        method: "POST",
        headers: { "X-Requested-With": "XMLHttpRequest" },
        body: expect.any(FormData),
      }
    )
    expect(offers).toMatchSnapshot()
  })
})
