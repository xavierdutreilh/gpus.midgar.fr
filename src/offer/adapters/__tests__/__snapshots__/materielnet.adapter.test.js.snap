// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOffers retrieves materiel.net offers 1`] = `
Array [
  Object {
    "key": "AR202009070050",
    "name": "MSI GeForce RTX 3090 Gaming X TRIO",
    "price": "€1,849.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070050.html",
  },
  Object {
    "key": "AR202009080079",
    "name": "Gigabyte GeForce RTX 3090 EAGLE OC 24G",
    "price": "€1,699.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009080079.html",
  },
  Object {
    "key": "AR202009010062",
    "name": "Asus GeForce RTX 3090 TUF",
    "price": "€1,849.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009010062.html",
  },
  Object {
    "key": "AR202009070051",
    "name": "MSI GeForce RTX 3090 Ventus 3X 24G",
    "price": "€1,729.94",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070051.html",
  },
  Object {
    "key": "AR202009090076",
    "name": "Asus GeForce RTX 3080 ROG STRIX OC",
    "price": "€1,099.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090076.html",
  },
  Object {
    "key": "AR202009090100",
    "name": "EVGA GeForce RTX 3080 FTW3 GAMING",
    "price": "€1,049.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090100.html",
  },
  Object {
    "key": "AR202009180077",
    "name": "Zotac GeForce RTX 3080 AMP HOLO",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009180077.html",
  },
  Object {
    "key": "AR202009030021",
    "name": "Zotac GeForce RTX 3080 TRINITY",
    "price": "€899.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030021.html",
  },
  Object {
    "key": "AR202009030072",
    "name": "Inno3D GeForce RTX 3080 iCHILL X4",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030072.html",
  },
  Object {
    "key": "AR202009180078",
    "name": "Zotac GeForce RTX 3080 TRINITY OC",
    "price": "€949.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009180078.html",
  },
  Object {
    "key": "AR202009030020",
    "name": "Zotac GeForce RTX 3090 TRINITY",
    "price": "€1,729.94",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030020.html",
  },
  Object {
    "key": "AR202009030069",
    "name": "Inno3D GeForce RTX 3090 iCHILL X3",
    "price": "€1,699.94",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030069.html",
  },
  Object {
    "key": "AR202009070039",
    "name": "PNY GeForce RTX 3090 XLR8 Gaming EPIC-X Metal",
    "price": "€1,729.99",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070039.html",
  },
  Object {
    "key": "AR202009010057",
    "name": "Asus GeForce RTX 3080 TUF",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009010057.html",
  },
  Object {
    "key": "AR202009030073",
    "name": "Inno3D GeForce RTX 3080 iCHILL X3",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030073.html",
  },
  Object {
    "key": "AR202009070041",
    "name": "PNY GeForce RTX 3080 XLR8 Gaming EPIC-X Metal",
    "price": "€899.99",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070041.html",
  },
  Object {
    "key": "AR202009070052",
    "name": "MSI GeForce RTX 3080 Gaming X TRIO",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070052.html",
  },
  Object {
    "key": "AR202009090099",
    "name": "EVGA GeForce RTX 3080 XC3 GAMING",
    "price": "€979.96",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090099.html",
  },
  Object {
    "key": "AR202009090101",
    "name": "EVGA GeForce RTX 3080 FTW3 ULTRA GAMING",
    "price": "€1,099.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090101.html",
  },
  Object {
    "key": "AR202009030068",
    "name": "Inno3D GeForce RTX 3090 iCHILL X4",
    "price": "€1,749.95",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030068.html",
  },
  Object {
    "key": "AR202009030081",
    "name": "Asus GeForce RTX 3090 TUF OC",
    "price": "€1,849.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030081.html",
  },
  Object {
    "key": "AR202009080064",
    "name": "Gigabyte GeForce RTX 3090 GAMING OC 24G",
    "price": "€1,769.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009080064.html",
  },
  Object {
    "key": "AR202009090106",
    "name": "EVGA GeForce RTX 3090 XC3 ULTRA GAMING",
    "price": "€1,899.95",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090106.html",
  },
  Object {
    "key": "AR202009030079",
    "name": "Inno3D GeForce RTX 3080 Twin X2 OC",
    "price": "€949.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030079.html",
  },
  Object {
    "key": "AR202009070038",
    "name": "PNY GeForce RTX 3090 XLR8 Gaming EPIC-X",
    "price": "€1,699.99",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070038.html",
  },
  Object {
    "key": "AR202009070040",
    "name": "PNY GeForce RTX 3080 XLR8 Gaming EPIC-X",
    "price": "€899.99",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070040.html",
  },
  Object {
    "key": "AR202009080083",
    "name": "Gigabyte GeForce RTX 3080 GAMING OC 10G",
    "price": "€999.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009080083.html",
  },
  Object {
    "key": "AR202009080086",
    "name": "Gigabyte GeForce RTX 3080 EAGLE OC 10G",
    "price": "€949.96",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009080086.html",
  },
  Object {
    "key": "AR202009090081",
    "name": "Asus GeForce RTX 3090 ROG STRIX OC",
    "price": "€1,949.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090081.html",
  },
  Object {
    "key": "AR202009090098",
    "name": "EVGA GeForce RTX 3080 XC3 BLACK",
    "price": "€949.96",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090098.html",
  },
  Object {
    "key": "AR202009090102",
    "name": "EVGA GeForce RTX 3090 XC3 BLACK",
    "price": "€1,739.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090102.html",
  },
  Object {
    "key": "AR202009090103",
    "name": "EVGA GeForce RTX 3090 XC3 GAMING",
    "price": "€1,829.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090103.html",
  },
  Object {
    "key": "AR202009090107",
    "name": "EVGA GeForce RTX 3090 FTW3 GAMING",
    "price": "€1,999.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090107.html",
  },
  Object {
    "key": "AR202009090109",
    "name": "EVGA GeForce RTX 3090 FTW3 ULTRA GAMING",
    "price": "€2,049.95",
    "status": "available",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009090109.html",
  },
  Object {
    "key": "AR202009100088",
    "name": "EVGA GeForce RTX 3080 XC3 ULTRA GAMING",
    "price": "€949.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009100088.html",
  },
  Object {
    "key": "AR202009030080",
    "name": "Asus GeForce RTX 3080 TUF OC",
    "price": "€1,049.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009030080.html",
  },
  Object {
    "key": "AR202009070053",
    "name": "MSI GeForce RTX 3080 Ventus 3X 10G",
    "price": "€949.96",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009070053.html",
  },
  Object {
    "key": "AR202009180076",
    "name": "Zotac GeForce RTX 3080 AMP EXTREME HOLO",
    "price": "€1,099.94",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009180076.html",
  },
  Object {
    "key": "AR202009290084",
    "name": "Gigabyte Aorus GeForce RTX 3080 MASTER 10G",
    "price": "€1,149.95",
    "status": "unavailable",
    "store": "materielnet",
    "url": "https://www.materiel.net/produit/202009290084.html",
  },
]
`;
